File: Semestrul3/map/examen/conditional/src/Model/Statement/CompoundStatement.java
package Model.Statement;

import Model.ADT.IStack;
import Model.ProgramState;

public class CompoundStatement implements IStmt {
    IStmt first;
    IStmt snd;
    public CompoundStatement(IStmt f, IStmt s)
    {
        this.first=f;
        this.snd=s;
    }

    @Override
    public ProgramState execute(ProgramState state)
    {
        IStack<IStmt> stk=state.getExeStack();
        stk.push(snd);
        stk.push(first);
        return null;
    }

    @Override
    public String toString()
    {
        return "("+first.toString() + ";" + snd.toString()+")";
    }
}

File: Licenta/Semester 3/Toy Language Interpreter/C# Implementation/Lab7/Model/Statements/CompoundStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Lab7.Model.ADT;
namespace Lab7.Model.Statements
{
    public class CompoundStatement : Statement
    {
        private Statement first, second;

        public CompoundStatement(Statement f, Statement s)
        {
            this.first = f;
            this.second = s;
        }

        public PrgState Execute(PrgState p)
        {
            IExeStack<Statement> stk = p.Stack;
            stk.Push(this.second);
            stk.Push(this.first);
            return p;
        }

        public override string ToString()
        {


             return "(" + first.ToString() + ";" + second.ToString() + ")";
           
        }
    }
}
